### PR TITLE
resuming checkpoint without lr schedule or optimizer state

### PR DESCRIPTION
**Pull request by eliebak**
quick pr to add the possibility to resume training without the optimizer state or lr schedule for continual learning

### CODE REVIEW

File: src/nanotron/trainer.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def __init__(
             )
 
         # Define iteration start state
-        if self.init_checkpoint_path is not None:
+        if self.init_checkpoint_path is not None and self.config.checkpoints.load_lr_scheduler:
```
Comment (NouamaneTazi) [inline src/nanotron/trainer.py:218]: is this only lr related?

**Review by NouamaneTazi — COMMENTED** [review]
Nice! left a qst

File: src/nanotron/trainer.py (hunk 1 above)
Comment (eliebak) [reply src/nanotron/trainer.py:218]: I think when you want to do a new lr schedule, you also want to do a different data stages + reset the "last_train_step" argument

**Review by NouamaneTazi — APPROVED** [review]
LGTM!
